translation-web/src/main/java/eu/europeana/api/translation/config/TranslationServiceConfigProvider.java

```diff
@@ -56,63 +57,57 @@ public Map<String, TranslationService> getTranslationServices() {
     return translationServices;
   }
 
-//  // executed after all beans are initialized, to initialize the services from the main json config
-//  // file
-//  @Override
-//  public void onApplicationEvent(ContextRefreshedEvent event) {
-//    try {
-//      initTranslationServicesConfiguration();
-//    } catch (Exception e) {
-//      throw new RuntimeException(
-//          "The initialization of the services from the global json config has failed.", e);
-//    }
-//  }
-
-  public void initTranslationServicesConfiguration() throws Exception {
-    readTranslationServicesConfig();
-    validateTranslationServicesConfig();
+  public void initTranslationServicesConfiguration(){
+    try {
+      // init translation services
+      readTranslationServicesConfig();
+      validateTranslationServicesConfig();
+    }catch(JsonProcessingException | ClassNotFoundException | LanguageDetectionException e) {
+      throw new RuntimeException("Invalid Service Configurations!", e);
+    }
   }
 
   private void validateTranslationServicesConfig()
-      throws TranslationException, ClassNotFoundException {
+      throws LanguageDetectionException, ClassNotFoundException {
 
     validateDetectServiceCfg();
     validateTranslationServiceCfg();
   }
 
-  private void validateTranslationServiceCfg() throws TranslationException, ClassNotFoundException {
+  private void validateTranslationServiceCfg()
```

Comment:
refractor this method to reduce cognitive complexity to 15 from 28

Reply:
will do in a follow up ticket for addressing quality issues